packages/bus/src/brokers/RMQEventBus.spec.ts

```diff
@@ -0,0 +1,550 @@
+          prop.toString()
+        )
+      ) {
+        return undefined;
+      }
+
+      return Reflect.get(target, prop, receiver);
+    }
+  });
+
+const randomString = (length: number): string => {
+  let result = '';
+  const characters =
+    'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  const charactersLength = characters.length;
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+  }
+
+  return result;
+};
+
+describe('RMQEventBus', () => {
+  let eventBus!: RMQEventBus;
+  let sdkConfig!: Configuration;
+  const busConfig = {
+    exchange: 'EventBus',
+    clientQueue: 'agent:nnCF9MfHpbvdJVtSbQfKa1',
+    connectTimeout: 10000,
+    heartbeatInterval: 5000,
+    reconnectTime: 5000
+  };
+  const mockedConnectionFactory = mock<ConnectionFactory<Connection>>();
+  const mockedConnection = mock<Connection>();
+  const mockedConfirmChannel = mock<ConfirmChannel>();
+  const retryStrategy = new ExponentialBackoffRetryStrategy({ maxDepth: 2 });
+
+  beforeEach(() => {
+    when(
+      mockedConfirmChannel.consume(anything(), anything(), anything())
+    ).thenResolve({ consumerTag: `agent:${randomString(22)}` } as any);
+
+    when(
+      mockedConfirmChannel.assertExchange(anything(), anything(), anything())
+    ).thenResolve();
+
+    when(
+      mockedConfirmChannel.assertQueue(anything(), anything())
+    ).thenResolve();
+
+    const confirmChannel = resolvableInstance(mockedConfirmChannel);
+    when(mockedConnection.createConfirmChannel()).thenResolve(
+      confirmChannel as any
+    );
+
+    const connection = resolvableInstance(mockedConnection);
+    when(mockedConnectionFactory.create(anything(), anything())).thenResolve(
+      connection
+    );
+
+    sdkConfig = new Configuration({
+      cluster: 'amqps://test-url.com:5672',
+      credentials: {
+        username: 'bot',
+        token: 'weobbz5.nexa.vennegtzr2h7urpxgtksetz2kwppdgj0'
+      }
+    });
+
+    eventBus = new RMQEventBus(
+      sdkConfig,
+      busConfig,
+      retryStrategy,
+      instance(mockedConnectionFactory)
+    );
+  });
+
+  afterEach(() =>
+    reset<ConnectionFactory<Connection> | Connection | ConfirmChannel>(
+      mockedConnectionFactory,
+      mockedConnection,
+      mockedConfirmChannel
+    )
+  );
+
+  describe('init', () => {
+    it('should init event bus', async () => {
+      await expect(eventBus.init?.call(eventBus)).resolves.not.toThrow();
+      verify(mockedConnectionFactory.create(anything(), anything())).once();
+      verify(mockedConnection.createConfirmChannel()).once();
+      verify(
+        mockedConfirmChannel.consume(anything(), anything(), anything())
+      ).twice();
+      verify(
+        mockedConfirmChannel.assertExchange(anything(), anything(), anything())
+      ).once();
+      verify(mockedConfirmChannel.assertQueue(anything(), anything())).once();
```

Comment:
You should verify only these expectation which really make sense for the test purpose. Moreover, expectations must be precision.